content/en/docs/tasks/configure-pod-container/configure-gmsa.md

```diff
@@ -0,0 +1,302 @@
+      - name: ${DEPLOYMENT_NAME}
+        image: k8ssigwindows/gmsa-admission-webhook
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 443
+        volumeMounts:
+          - name: tls
+            mountPath: "/tls"
+            readOnly: true
+        env:
+          - name: TLS_KEY
+            value: /tls/key
+          - name: TLS_CRT
+            value: /tls/crt
+      volumes:
+      - name: tls
+        secret:
+          secretName: ${DEPLOYMENT_NAME}
+          items:
+          - key: tls_private_key
+            path: key
+          - key: tls_certificate
+            path: crt
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ${DEPLOYMENT_NAME}
+  namespace: ${NAMESPACE}
+spec:
+  ports:
+  - port: 443
+    targetPort: 443
+  selector:
+    app: ${DEPLOYMENT_NAME}
+---
+# add a label to the deployment's namespace so that we can exclude it
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: ${NAMESPACE}
+  labels:
+    gmsa-webhook: disabled
+---
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: ${DEPLOYMENT_NAME}
+webhooks:
+- name: k8s-gmsa-admission-webhook.sig-windows.k8s.io
+  clientConfig:
+    service:
+      name: ${DEPLOYMENT_NAME}
+      namespace: ${NAMESPACE}
+      path: "/validate"
+    caBundle: ${CA_BUNDLE}
+  rules:
+  - operations: ["CREATE", "UPDATE"]
+    apiGroups: [""]
+    apiVersions: ["*"]
+    resources: ["pods"]
+  failurePolicy: Fail
+  # don't run on ${NAMESPACE}
+  namespaceSelector:
+    matchExpressions:
+      - key: gmsa-webhook
+        operator: NotIn
+        values: [disabled]
+---
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: ${DEPLOYMENT_NAME}
+webhooks:
+- name: k8s-gmsa-admission-webhook.sig-windows.k8s.io
+  clientConfig:
+    service:
+      name: ${DEPLOYMENT_NAME}
+      namespace: ${NAMESPACE}
+      path: "/mutate"
+    caBundle: ${CA_BUNDLE}
+  rules:
+  - operations: ["CREATE"]
+    apiGroups: [""]
+    apiVersions: ["*"]
+    resources: ["pods"]
+  failurePolicy: Fail
+  # don't run on ${NAMESPACE}
+  namespaceSelector:
+    matchExpressions:
+    - key: gmsa-webhook
+      operator: NotIn
+      values: [disabled]
+```
+
+### Configure cluster role to enable RBAC on specific GMSA credential specs
+A cluster role needs to be defined for each GMSA that authorizes the `use` verb on a specific GMSA resource by a subject such as a service account. The following shows an example of a cluster role that authorizes usage of gmsa-WebApp1 credspec from above. Save the file as gmsa-webapp1-role.yaml and apply using `kubectl apply -f gmsa-webapp1-role.yaml`
+
+```
+#Create the Role to read the credspec
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: my-rbac-reader
+rules:
+- apiGroups: ["windows.k8s.io"]
+  resources: ["gmsacredentialspecs"]
+  verbs: ["use"]
+  resourceNames: ["gmsa-WebApp1"]
```

Comment:
Could be worth highlighting that this is the name of the GMSA cred spec resource created earlier, and that you want to expose to your deployment?

Reply:
done